Program.cs

```diff
@@ -1,2 +1,62 @@
-﻿// See https://aka.ms/new-console-template for more information
-Console.WriteLine("Hello, World!");
+﻿using System.Text;
+
+namespace SerializationDeserialization;
+
+internal class Program
+{
+    static void Main()
+    {
+        Person person = new Person() { Name = "Alex", Surname = "Great", Born = 356 };
+
+        SerializePersonProperties(person, "alex_great2.txt");
+    }
+
+    static Dictionary<string, string> ReadDataFile(string path)
+    {
+        string[] data = File.ReadAllLines(path);
+        Dictionary<string, string> properties = new Dictionary<string, string>();
+
+        for (int i = 0; i < data.Length; i++)
+        {
+            string[] splittedData = data[i].Split(':');
+            properties.Add(splittedData[0], splittedData[1]);
+        }
+
+        return properties;
+    }
+
+    static void SetPersonProperties(Person person, Dictionary<string, string> data)
+    {
+        var propertyInfos = person.GetType().GetProperties();
+
+        for (int i = 0; i < propertyInfos.Length; i++)
+        {
+            if (data.ContainsKey(propertyInfos[i].Name))
+            {
+                try
+                {
+                    Type propertyType = propertyInfos[i].PropertyType;
+                    var convertedValue = Convert.ChangeType(data[propertyInfos[i].Name], propertyType);
+                    propertyInfos[i].SetValue(person, convertedValue, null);
+                }
+                catch (FormatException ex)
+                {
+                    Console.WriteLine(ex.Message);
```

Comment:
There is also an important thing to know about `throw` and `throw ex` when re-throwing exception from the `catch` block.